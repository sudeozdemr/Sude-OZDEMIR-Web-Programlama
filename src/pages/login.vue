<template>
  <div>
    <ust-menu></ust-menu>
    <uye-girisi/>
  </div>
</template>
<script>
import { FontAwesomeIcon } from "@fortawesome/vue-fontawesome";
import { library } from "@fortawesome/fontawesome-svg-core";
import { fas } from "@fortawesome/free-solid-svg-icons";
import ustMenu from "../components/ustMenu.vue";
import { mapActions } from "vuex";

import { auth } from "../../firebase/config"
import UyeGirisi from '../components/uyeGirisi.vue';
library.add(fas);
export default {
  components: {
    FontAwesomeIcon,
    ustMenu,
    UyeGirisi,
  }
};
</script>