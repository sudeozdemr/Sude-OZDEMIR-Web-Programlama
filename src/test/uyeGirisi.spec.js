import { mount, createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'
import login from '../components/uyeGirisi.vue'

import storeConfig from "../store";

const localVue = createLocalVue();
localVue.use(Vuex);
describe('login', () => {
    let wrapper;
    const mockFunction = jest.fn();
    describe('login Butonuna Basıldığında', () => {
        beforeEach(() => {
            wrapper = mount(login, {
                localVue,
                data() {
                    return {
                        userInfo: {
                            email: 'test@test.com',
                            password: '111222',
                        }
                    };
                },
                store: new Vuex.Store({
                    actions: {
                        login: mockFunction,
                    },
                }),
                mocks: {
                    $router: {
                        push: jest.fn()
                    }
                },
                stubs: ['router-link']
            });
        });
        // eslint-disable-next-line require-await
        test('login fonksiyonun çağrılması', async() => {
            const loginButton = wrapper.find(".login_button");
            loginButton.trigger("click").then(() => {
                expect(mockFunction).toHaveBeenCalled();
            });
        });
    });

    test('Vue Örneği mi?', () => {
        const wrapper = mount(login, { stubs: ['router-link'] })
        expect(wrapper.vm).toBeTruthy()
    });

    test("Snapshot Testi", () => {
        expect(wrapper.html()).toMatchSnapshot();
    });
});