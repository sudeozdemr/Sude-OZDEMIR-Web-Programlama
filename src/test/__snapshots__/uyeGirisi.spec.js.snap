// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login Snapshot Testi 1`] = `
"<div class=\\"container\\">
  <div class=\\"uyelikHeader\\">
    <div class=\\"logoBox\\"><img src=\\"src/assets/uyegirisiprofil.png\\" alt=\\"yok\\" style=\\"width: 60px; margin-top: 7px;\\"></div>
    <h2>Üye Girişi</h2>
  </div>
  <div class=\\"uyariBox\\">
    <p>Lütfen üye girişi yapınız</p>
  </div>
  <form><label>E-posta adresiniz</label> <input type=\\"text\\" required=\\"required\\"> <label>Şifreniz</label> <input type=\\"password\\" required=\\"required\\"> <label>Doğrula</label>
    <div class=\\"reCaptcha\\"><label class=\\"beniHatirla\\"><input type=\\"checkbox\\">Ben robot değilim
      </label> <img src=\\"src/assets/recaptcha.png\\"></div> <label class=\\"beniHatirla\\"><input type=\\"checkbox\\">BENİ HATIRLA
    </label> <input type=\\"submit\\" value=\\"GİRİŞ YAP\\" class=\\"login_button\\"> <a href=\\"#\\" class=\\"passwordLink\\"> Şifremi Unuttum</a>
  </form>
</div>"
`;
